### PR TITLE
Dblatcher check rendering options before launch

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?

The launch wizard will check if the the draft is "article-based" and has missing or incomplete `renderingOptions`. If so, the UI will display appropriate markdown and not allow the user to proceed with the launch.

Adds a new category of "article-based-legacy" to represent the newsletters that are 'article-based' but not rendered in email rendering. We don't expect any new ones to be created, but having this category will allow use to more accurately represent the current newsletters and track which ones need "re-launching". 

https://trello.com/c/aFgiLzRU/393-when-validating-whether-able-to-launch-for-article-based-newsletters-also-need-to-check-all-mandatory-rendering-options-have-bee

## How to test

Attempt to launch the 'Avon Recumbent' draft. The launch wizard will direct you to complete the rendering options wizard. When you have done this, the launch wizard will allow you to launch. 

## How can we measure success?

Better validation / work flow. Avoid missing set up data needed for email-rendering.

## Images

Screenshot 2023-05-04 at 14 05 32

### CODE REVIEW

**Review by phillipbarron — COMMENTED** [review]
Post resolving conflict this looks good.
In another PR, perhaps we might look at direct linking to the park to the form that is relevant

**Comment by dblatcher** [comment]
> Post resolving conflict this looks good. In another PR, perhaps we might look at direct linking to the park to the form that is relevant

I was thinking that on wizards you can skip in, it might be a nice feature for the page to support a search param for which step to start on - eg /newsletters/newsletter-data/7001?step=launch-date

**Review by phillipbarron — APPROVED** [review]
Looks good - left a small note regarding nested ternary 👍

File: libs/newsletter-workflow/src/lib/steps/launchNewsletter/isDataCompleteLayout.ts [hunk 1]
```diff
@@ -14,29 +15,48 @@ const markdownTemplate = `
...
 const staticMarkdown = markdownTemplate.replace(
 	regExPatterns.name,
 	'the newsletter',
 );
 
+const getReadinessMessage = (
+	hasAllStandardData?: boolean,
+	hasRenderingOptionsIfNeeded?: boolean,
+) => {
+	return hasAllStandardData
```
[the remark this answers is not in the document]
Comment (dblatcher) [reply libs/newsletter-workflow/src/lib/steps/launchNewsletter/isDataCompleteLayout.ts]: 👍  done - I updated property names in `markdownAnswers` to be more explicit too.

**Pull request merged**